src/types.ts

```diff
@@ -1,17 +1,24 @@
 import React from 'react';
 import { FieldErrors, SubmitErrorHandler, SubmitHandler } from 'react-hook-form';
-import { DataSourceSettings, OrgRole, SelectableValue } from '@grafana/data';
+import { SelectableValue } from '@grafana/data';
 import { EmbeddedScene, SceneRouteMatch } from '@grafana/scenes';
 import { DataSourceRef } from '@grafana/schema';
 import { ZodType } from 'zod';
 
-import { LinkedDatasourceInfo } from './datasource/types';
+import { LinkedDatasourceInfo, ProvisioningLinkedDatasourceInfo } from './datasource/types';
 import { FilterType } from 'hooks/useCheckFilters';
 import { Assertion, MultiHttpEntry, MultiHttpVariable, RequestProps } from 'components/MultiHttp/MultiHttpTypes';
 
-export interface GlobalSettings {
+export interface ProvisioningJsonData {
   apiHost: string;
-  stackId?: number;
+  stackId: number;
+  metrics: ProvisioningLinkedDatasourceInfo;
+  logs: ProvisioningLinkedDatasourceInfo;
+}
+
+export interface InitializedJsonData {
+  apiHost: string;
+  stackId: number;
   metrics: LinkedDatasourceInfo;
   logs: LinkedDatasourceInfo;
 }
```

Comment:
I found when I looked into it the `GlobalSettings` type was a **big** source of confusion for me. I couldn't work out why sometimes we had the linked datasource's uid and sometimes we didn't.

It is because there is our provisioning file 'global settings' which only requires the grafana name and hostedId inputted by the user for each datasource and after that is submitted our call to the Synthetic Monitoring api takes over and does a look up using that information and returns the `uid` (and type) for us.

So the reality is they are different types used for different purposes and should be typed separately to one another.